src/gpu_sysman_test.c

```diff
@@ -755,40 +755,75 @@ static int validate_and_reset_saved_metrics(unsigned int base_rounds,
   return missing + wrong;
 }
 
+/* Add family name + unit to the given buffer, return offset to the end
+ */
+static size_t add_family_name(char *buf, size_t bufsize,
+                              metric_family_t const *fam) {
+  static const struct {
+    const char *unit;
+    const char *name;
+  } units[] = {
+      {"1", "_ratio"},     {"By", "_bytes"}, {"By/s", "_bytes_per_second"},
+      {"Cel", "_celsius"}, {"{error}", ""},  {"MHz", "_mhz"},
+      {"uj", "_ujoules"},  {"us", "_usecs"}, {"W", "_watts"},
```

Comment:
That the code needs to add a suffix based on the unit is an indicator that the metric names are not sufficiently unique. Note that most systems used to aggregate metrics will not be able to filter/group by the unit.

Reply:
Thanks, that's good to know!   I'll drop that part of the tester if all metrics get satisfactory unique names.